packages/core/src/hydration/annotate.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ApplicationRef} from '../application_ref';
+import {TNode} from '../render3/interfaces/node';
+import {isLContainer} from '../render3/interfaces/type_checks';
+import {HEADER_OFFSET, HOST, LView, TVIEW} from '../render3/interfaces/view';
+import {unwrapRNode} from '../render3/util/view_utils';
+import {makeStateKey, TransferState} from '../transfer_state';
+
+import {TRANSFER_STATE_TOKEN_ID} from './api';
+import {NghDom} from './interfaces';
+import {getComponentLView, NGH_ATTR_NAME} from './utils';
+
+/**
+ * Refers to the hydration data in the transferred state.
+ */
+export const NGH_DATA_KEY = makeStateKey<Array<NghDom>>(TRANSFER_STATE_TOKEN_ID);
+
+/**
+ * Keeps track of all produced `ngh` annotations and avoids
+ * duplication. If the same annotation is being added, the collection
+ * remains the same and an index of that annotation is returned instead.
+ * This helps minimize the amount of annotations needed on a page.
+ */
```

Comment:
```suggestion
/**
 * A collection that tracks all `ngh` annotations to avoid
 * duplication. If attempting to add a duplicate annotation, the collection
 * returns the index of the previously recorded annotation. This reduces 
 * the number of annotations needed for a given page.
 */
```

nit: a good rule of thumb is that that JsDoc descriptions for classes should follow from "ClassName is...", e.g. "NgcAnnotationCollection is..." (similar for other class and interface descriptions in this PR)

(compared to functions which generally start with verbs)

Reply:
Updated! Thanks!